File: MarkdownComponent/ExampleJsInterop.cs
using System;
using Microsoft.AspNetCore.Blazor.Browser.Interop;

namespace MarkdownComponent
{
    public class ExampleJsInterop
    {
        public static string Prompt(string message)
        {
            return RegisteredFunction.Invoke<string>(
                "MarkdownComponent.ExampleJsInterop.Prompt",
                message);
        }
    }
}

File: MarkdownComponent/content/exampleJsInterop.js
// This file is to show how a library package may provide JavaScript interop features
// wrapped in a .NET API

Blazor.registerFunction('MarkdownComponent.ExampleJsInterop.Prompt', function (message) {
   
    let el = document.getElementById("markdown-component");
    if (el) { el.innerHTML = message; }
    else {
        console.log("HTML not rendered");
    }
});
